applications/nrf_desktop/src/modules/fast_pair.c

```diff
@@ -46,20 +48,35 @@ static bool in_pairing_mode(uint8_t bt_local_id)
 	return (bond_cnt(bt_local_id) == 0);
 }
 
-static void update_fast_pair_ui_indication(void)
+static void update_fast_pair_ui_indication(uint8_t bt_local_id)
```

Comment:
This looks like the opposite stance of the one you suggested in one of the PRs:

https://github.com/nrfconnect/sdk-nrf/pull/10066#discussion_r1091989968
(your drive to make the function stateless)

Are you sure about this?

Reply:
The approach depends on use-case. In some cases it's better to make functions stateless and in some cases it's better not to do it.

In the linked case, the function simply generates advertising mode relying on passed arguments. The advertising mode depends only on the 2 passed arguments and we in general can check the advertising mode for any combination of these arguments.

Here, the function should update showing Fast Pair Provider's UI indication relying on state of the selected Bluetooth local identity and application local identity. `ble_peer_operation_event` updates identities in use, other events, like `ble_peer_event` only trigger update. I suggested to split modification of used identities from only triggering update for Fast Pair Provider also to prevent issues similar to one where identity from `ble_peer_event` informing about disconnection was accidentally used.

Still I would not state that changing this one is hard requirement.

Reply:
ok